core/drivers/stm32_gpio.c

```diff
@@ -638,12 +733,48 @@ static TEE_Result stm32_gpio_get_dt(struct dt_pargs *pargs, void *data,
 		return TEE_ERROR_GENERIC;
 	}
 
+	if (gpio->dt_flags & GPIO_STM32_NSEC)
+		gpio_secure = false;
+
 	state = calloc(1, sizeof(*state));
 	if (!state) {
 		free(gpio);
 		return TEE_ERROR_OUT_OF_MEMORY;
 	}
 
+	SLIST_FOREACH(reg_state, &consumed_gpios_head, link) {
+		if (reg_state->gpio_pinctrl.bank == bank->bank_id &&
+		    reg_state->gpio_pinctrl.pin == gpio->pin) {
+			EMSG("node %s: GPIO %c%u is used by another device",
+			     consumer_name, bank->bank_id + 'A', gpio->pin);
+			free(state);
+			free(gpio);
+			return TEE_ERROR_GENERIC;
+		}
+	}
+
+	if (!pin_is_accessible(bank, gpio->pin)) {
```

Comment:
Kind of the same fight here. I think we should "acquire" the access when consuming the GPIO. That probably is here or in each of the exposed APIs:
```
	.get_direction = stm32_gpio_get_direction,
	.set_direction = stm32_gpio_set_direction,
	.get_value = stm32_gpio_get_level,
	.set_value = stm32_gpio_set_level,
```
but we also need to release the access when we put the GPIO in stm32_gpio_put_gpio().

We don't have a pinctrl release

Reply:
I would expect GPIOs consumed by OP-TEE are only owned by OP-TEE. We have no current plan on stm32mpX platforms to share GPIOs at runtime.